lib/menu.rb

```diff
@@ -0,0 +1,36 @@
+class Menu
+
+  attr_reader :name
+
+  def initialize(name, *dishes)
+    @name = name
+    @dishes = Hash.new
+    load_dishes(dishes)
+  end
+
+  def dishes
+    @dishes.dup
+  end
+
+  def to_s
+    string = "#{name} Menu\n"
+    @dishes.each do |dish, price|
+      string += "#{dish}".ljust(25)
+      string += "#{'%.02f' % price}".rjust(10)
+      string += "\n"
+    end
+    string
+  end
+
+  private
+
+  def load_dishes(dishes)
+    fail "No dishes in the menu" if dishes.empty?
+    dishes.each_slice(2) do |name, price|
+      fail "Invalid dish name '#{name}'" unless name.is_a?(String)
```

Comment:
Validation is a good thing to think about, but validate for things that are important - you can check an object 'responds_to' any important methods, or you can validate that a necessary object is present, for example.